cEP-0027.md

```diff
@@ -0,0 +1,149 @@
+             If duration > warning_time:
+                 print(duration)
+                 sys.stdout.flush()
+     ```
+
+### 2. Implementation of `GlobalBearTestHelper` class
+
+Currently, many methods are redundant in the Global Bear tests. For example:
+
+- 1
+
+  ```
+  def get_absolute_test_path(file):
+      return path/of/test_file/file
+  ```
+
+- 2
+
+  ```
+  def get_results(self, files_to_check):
+      self.file_dict = [get_absolute_test_path(files_to_check)]
+      self.uut = AnyGlobalBear(self.file_dict, self.section, self.queue)
+      return list(self.uut.run())
+  ```
+
+- 3
+
+  ```
+  def test_results(self):
+      results = self.get_results([self.test_files1, self.test_files2])
+      messages = [result.message for result in results]
+      self.assertEquals(messages, [“This is message 1.”,
+                                   “This is message 2.”])
+  ```
+
+Here is the prototype implementation of these repeating methods into
+`GlobalBearTestHelper` by inheriting [pytest](https://pytest.org/),
+[unittest.TestCase](https://docs.python.org/3/library/unittest.html),
+[GlobalBear](http://api.coala.io/en/latest/coalib.bears.html#module-coalib.bears.GlobalBear):
+
+```
+    def get_test_path(test_dir, file):
+        return os.path.join(os.path.dirname(__file__), test_dir, file)
+
+    def get_results(global_bear, test_dir, file_dict, settings={}):
+        for file in file_dict:
+            with get_test_path(test_dir, file) as fname,
+                execute_bear(global_bear, fname, **settings) as bear_output:
+                return bear_output
+
+    @contextmanager
+    def execute_bear(bear, *args, **kwargs):
+        try:
+            bear_output_generator = bear.execute(*args, **kwargs)
+            assert bear_output_generator is not None, \
+                'Bear returned None on execution\n'
+            yield bear_output_generator
+        except Exception as err:
+            msg = []
+            while not bear.message_queue.empty():
+                msg.append(bear.message_queue.get().message)
+            raise AssertionError(str(err) + ' \n' + '\n'.join(msg))
+        return list(bear_output_generator)
+
+
+    class GlobalBearTestHelper(unittest.TestCase)
+        def check_results(self, global_bear, test_dir, file_dict, results,
+                          settings={}):
+            if results in [[], ()]:
+                msg = ("The global bear '{}' yields a result although "
+                       "it shouldn't.".format(global_bear.__class__.__name__))
+
+            else:
+                msg = ("The global bear '{}' doesn't yield the right "
+                       "results.".format(global_bear.__class__.__name__))
+
+            bear_output = get_results(global_bear,
+                                      test_dir=test_dir,
+                                      file_dict=file_dict,
+                                      settings=settings)
+
+            self.assertEqual(bear_output, results, msg=msg)
```

Comment:
If you change this to `assert bear_output == results, msg`, you can remove the unittest dependency. Although `unittest`-style tests are used in most places in this code base, we use pytest as a test runner and I would advocate writing the more modern pytest style tests in the future.

Reply:
I have to look for it. Any reference suggested? @NiklasMM